src/earn/EarnEnterAmount.tsx

```diff
@@ -202,30 +228,45 @@ function EarnEnterAmount({ route }: Props) {
         localAmount: parsedLocalAmount,
       }
     }
-  }, [tokenAmountInput, localAmountInput, enteredIn, token])
+  }, [tokenAmountInput, localAmountInput, enteredIn, transactionToken, mode])
 
-  const feeCurrencies = useSelector((state) => feeCurrenciesSelector(state, token.networkId))
+  const feeCurrencies = useSelector((state) =>
+    feeCurrenciesSelector(state, transactionToken.networkId)
+  )
+
+  const balanceInTransactionToken = useMemo(
```

Comment:
isn't this `balanceInInputToken`? Seems like you're taking the tx token and converting to input token